passportAuth.go

```diff
@@ -0,0 +1,191 @@
+package gowebdav
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+// PassportAuth structure holds our credentials
+type PassportAuth struct {
+	user       string
+	pw         string
+	partnerUrl string
+	header     http.Header
+	cookies    []http.Cookie
+}
+
+// constructor for PassportAuth creates a new PassportAuth object and
+// automatically authenticates against the server
+func NewPassportAuth(user, pw, reqUrl string, header *http.Header) (*PassportAuth, error) {
+	p := &PassportAuth{
+		user:       user,
+		pw:         pw,
+		partnerUrl: reqUrl,
+		header:     header.Clone(),
+	}
+	err := p.genCookies()
+	if err != nil {
+		return nil, err
+	}
+	return p, nil
+}
+
+// Authorize the current request
+func (p *PassportAuth) Authorize(c *http.Client, rq *http.Request, path string) error {
+	// prevent redirects to detect subsequent authentication requests
+	rq.Header.Set(XInhibitRedirect, "1")
+	for _, cookie := range p.cookies {
+		rq.AddCookie(&cookie)
+	}
+	return nil
+}
+
+// Verify verifies if the authentication is good
+func (p *PassportAuth) Verify(c *http.Client, rs *http.Response, path string) (redo bool, err error) {
+	if rs.StatusCode == 302 {
+		if rs.Header.Get("WWW-Authenticate") != "" {
+			// re-authentication required as we are redirected to the login page
```

Comment:
this case should handle the ping pong with the partner server from `rs.Location`